RecoParticleFlow/PFProducer/src/PFEGammaFilters.cc

```diff
@@ -10,76 +10,83 @@
-  ele_maxE(ele_protectionsForJetMET.getParameter<double>("maxE")),
-  ele_maxEleHcalEOverEcalE(ele_protectionsForJetMET.getParameter<double>("maxEleHcalEOverEcalE")),
-  ele_maxEcalEOverPRes(ele_protectionsForJetMET.getParameter<double>("maxEcalEOverPRes")), 
-  ele_maxEeleOverPoutRes(ele_protectionsForJetMET.getParameter<double>("maxEeleOverPoutRes")),
-  ele_maxHcalEOverP(ele_protectionsForJetMET.getParameter<double>("maxHcalEOverP")), 
-  ele_maxHcalEOverEcalE(ele_protectionsForJetMET.getParameter<double>("maxHcalEOverEcalE")), 
-  ele_maxEcalEOverP_1(ele_protectionsForJetMET.getParameter<double>("maxEcalEOverP_1")),
-  ele_maxEcalEOverP_2(ele_protectionsForJetMET.getParameter<double>("maxEcalEOverP_2")), 
-  ele_maxEeleOverPout(ele_protectionsForJetMET.getParameter<double>("maxEeleOverPout")), 
-  ele_maxDPhiIN(ele_protectionsForJetMET.getParameter<double>("maxDPhiIN")),
-  badHcal_eleEnable_(ele_protectionsForBadHcal.getParameter<bool>("enableProtections")),
-  badHcal_phoTrkSolidConeIso_offs_(ph_protectionsForBadHcal.getParameter<double>("solidConeTrkIsoOffset")),
-  badHcal_phoTrkSolidConeIso_slope_(ph_protectionsForBadHcal.getParameter<double>("solidConeTrkIsoSlope")),
-  badHcal_phoEnable_(ph_protectionsForBadHcal.getParameter<bool>("enableProtections")),
-  debug_(false)
-{
-    readEBEEParams_(ele_protectionsForBadHcal, "full5x5_sigmaIetaIeta", badHcal_full5x5_sigmaIetaIeta_);
-    readEBEEParams_(ele_protectionsForBadHcal, "eInvPInv", badHcal_eInvPInv_);
-    readEBEEParams_(ele_protectionsForBadHcal, "dEta", badHcal_dEta_);
-    readEBEEParams_(ele_protectionsForBadHcal, "dPhi", badHcal_dPhi_);
-}
+namespace {
 
-void PFEGammaFilters::readEBEEParams_(const edm::ParameterSet &pset, const std::string &name, std::array<float,2> & out) {
+  void readEBEEParams_( const edm::ParameterSet &pset,
+                        const std::string &name,
+                        std::array<float,2> & out)
+  {
     const auto & vals = pset.getParameter<std::vector<double>>(name);
-    if (vals.size() != 2) throw cms::Exception("Configuration") << "Parameter " << name << " does not contain exactly 2 values (EB, EE)\n";
+    if (vals.size() != 2) throw cms::Exception("Configuration") << "Parameter "
+        << name << " does not contain exactly 2 values (EB, EE)\n";
     out[0] = vals[0]; 
     out[1] = vals[1]; 
+  }
+
+  auto eleJetMetProt(const edm::ParameterSet& cfg) {
+      return cfg.getParameter<edm::ParameterSet>("electron_protectionsForJetMET");
+  }
+
+  auto phJetMetProt(const edm::ParameterSet& cfg) {
+      return cfg.getParameter<edm::ParameterSet>("photon_protectionsForJetMET");
+  }
+
+  auto eleHcalProt(const edm::ParameterSet& cfg) {
+      return cfg.getParameter<edm::ParameterSet>("electron_protectionsForBadHcal");
+  }
+
+  auto phHcalProt(const edm::ParameterSet& cfg) {
+      return cfg.getParameter<edm::ParameterSet>("photon_protectionsForBadHcal");
```

Comment:
are the repeated calls to this and above optimized away?
If not, perhaps some initialization just has to be moved to the constructor body (or related members grouped into a single initializable member.

Reply:
I just moved it to the constructor body. That means no `const` private members in that case, but it doesn't matter in this case I'd say since all member functions are `const`.